### PR TITLE
Distinguish 0-size from size-unavailable

### PR DESCRIPTION
**Pull request by Cyan4973**
This patch introduces an error code for `UTIL_getFileSize()`, called `UTIL_FILESIZE_UNKNOWN`.

Up to now, `UTIL_getFileSize()` would answer `0` to mean "error", making it impossible to separate from a genuine empty file.

Most users of `UTIL_getFileSize()` were hard-wired to consider `0` as "error", so this change introduces many corrections throughout the code base.

0-size files are now correctly reported (header includes the `0` content size), while non-files (pipes) continue having an unknown size.

Is this functionality worth its complexity ?
Not sure, though I like the fact that checking for a file size error is now more explicit.

### CODE REVIEW

**Comment by Cyan4973** [comment]
Note : the visual studio test cannot be successful, as it currently uses the `ZSTD_MULTITHREAD` code path, which is [unable to express `srcSize==0`](https://github.com/facebook/zstd/blob/dev/lib/compress/zstdmt_compress.h#L53).

File: programs/fileio.c [hunk 1]
```diff
@@ -564,7 +564,7 @@ static unsigned long long FIO_compressGzFrame(cRess_t* ress,
                 strm.avail_out = (uInt)ress->dstBufferSize;
             }
         }
-        if (!srcFileSize)
+        if (srcFileSize != UTIL_FILESIZE_UNKNOWN)
```
Comment (terrelln) [inline programs/fileio.c]: `srcFileSize == UTIL_FILESIZE_UNKNOWN` here and below.

File: programs/fileio.c [hunk 2]
```diff
@@ -851,13 +850,13 @@ static int FIO_compressFilename_internal(cRess_t ress,
                 compressedfilesize += outBuff.pos;
         }   }
         if (g_nbThreads > 1) {
-            if (!fileSize)
+            if (fileSize != UTIL_FILESIZE_UNKNOWN)
```
Comment (terrelln) [inline programs/fileio.c]: `==`

File: programs/fileio.c (hunk 1 above)
Comment (Cyan4973) [reply programs/fileio.c]: Nice catch !

**Pull request merged**